packages/core/lib/fury.js

```diff
@@ -244,6 +256,33 @@ class Fury {
     return promise;
   }
 
+  /**
+   * Serialize synchronously an API Description into the given output format.
+   *
+   * @param {Object} options
+   * @param {Category} options.api
+   * @param {string} [options.mediaType]
+   */
+  serializeSync({ api, mediaType = 'text/vnd.apiblueprint' }) {
+    const adapter = findAdapter(this.adapters, mediaType, 'serialize');
+
+    if (!adapter) {
+      // eslint-disable-next-line no-console
+      console.warn('Media type did not match any registered serializer!');
```

Comment:
Should we make this throw so that it behaves the same way as the async counterpart?

```js
try {
    const result = await fury.serialize(...);
} catch (error) {
    //
}
```

```js
try {
    const result = fury.serializeSync(...);
} catch (error) {
    //
}
```

That way API consumers can catch the error and know the reason for failure programmatically.

Reply:
I thought about that. Just that if someone consumes it without wrapping in `try/catch` then it will terminate the control flow. It's not the equivalent of `Promise.reject`, also not something I would expect from this function.

In other hand the `console.warn` isn't nice either. I'm fine with changing as you suggest :)

Reply:
Addressed in 41b4d5f